From 8c39ce243708c3a3fe290154c5b85b093fadca5f Mon Sep 17 00:00:00 2001
From: Simon Ser <contact@emersion.fr>
Date: Sat, 27 Jan 2024 13:14:04 +0100
Subject: [PATCH] Drop automatic DEBUG define

We don't use the historical odd stable release numbering scheme
anymore.

Developers can still enable this debugging code via CFLAGS=-DDEBUG.

Signed-off-by: Simon Ser <contact@emersion.fr>
Closes: https://gitlab.freedesktop.org/pixman/pixman/-/issues/87
---
 pixman/pixman-private.h | 19 -------------------
 1 file changed, 19 deletions(-)

diff --git a/pixman/pixman-private.h b/pixman/pixman-private.h
index f43e87f..34fb69b 100644
--- a/pixman/pixman-private.h
+++ b/pixman/pixman-private.h
@@ -1050,28 +1050,9 @@ float pixman_unorm_to_float (uint16_t u, int n_bits);
  * Various debugging code
  */
 
-#undef DEBUG
-
 #define COMPILE_TIME_ASSERT(x)						\
     do { typedef int compile_time_assertion [(x)?1:-1]; } while (0)
 
-/* Turn on debugging depending on what type of release this is
- */
-#if (((PIXMAN_VERSION_MICRO % 2) == 0) && ((PIXMAN_VERSION_MINOR % 2) == 1))
-
-/* Debugging gets turned on for development releases because these
- * are the things that end up in bleeding edge distributions such
- * as Rawhide etc.
- *
- * For performance reasons we don't turn it on for stable releases or
- * random git checkouts. (Random git checkouts are often used for
- * performance work).
- */
-
-#    define DEBUG
-
-#endif
-
 void
 _pixman_log_error (const char *function, const char *message);
 
-- 
GitLab

